### ISSUE TITLE
Namespace categories (or the like)

### ISSUE THREAD

**Issue by chrislloyd** [chrislloyd/corkboard#16]
I've been thinking about how to group namespaces quite a bit. The general idea is being able to add a second level of namespaces can be really useful for navigation and comprehension. I've seen a bunch of examples of this around the internet. More specifically, it would be great to have this feature for the BRIO specs which lay things out in "Basics", "Components" & "Layouts".

Implementing this feature isn't the hard part, it's mostly the design of how it should work for the developer. I've settled on a couple of rules:

1. We shouldn't support arbitrary level trees of navigation. This gets difficult to display and difficult to design an interface for. It doesn't seem like it's needed either.
2. We should keep the "REPL for JS" principle.

Rule #1, is fairly straightforward, but rule #2 might need some explaining. A few different syntaxes I played around with were:

```
// creating groups at when importing cards:
devcards({
  Basics: [require('Colors/Colors-doc'), ... ],
  Components: [ ... ],
  Layouts: [ ... ],
});
```

The problem with this approach is that it felt like extra configuration. It's another file you need to edit when you create a new component.

```
// adding arguments or different syntaxes to `ns`:
ns('Basics/Colors');
ns('Basics', 'Colors');
ns('Colors', { group: 'Basics' });
```

This felt, weird because "groups" should be optional and the developer should have control over the order in which `ns`s' appear.

The approach I like this most is:

```
group('Basics');
ns('Colors');
card( ... );
```

This feels most "repl" like, and adopts a similar API to the `ns` and `card`. If I wanted to, I could put all my documentation in one big file and split it up however I wanted and the output would be obvious.

At the point, the biggest decision I think is what to call the function. I've already heard feedback that `ns` isn't obvious to new JS developers, which is understandable because it's cribbed from Clojure where it has actual semantic meaning. I think there's the opportunity to rename both functions (`group` and `ns`) to things that are more obvious to new developers but I suck at naming things. @jakobo @KevinGrandon @christianvuerings @juliacochran any suggestions?

**Comment by KevinGrandon** [comment]
I personally like `ns('Basics.Colors')`, which just feels familiar to me from past experience with namespacing.

I'm thinking I would want something like:
```
ns('Grid.Flexible');
ns('Grid.Classic');
ns('Grid.MultiColumn');
```

**Comment by chrislloyd** [comment]
Spaces gets ugly with using a dot. Also doesn't solve the issue of NS not being a JS concept.

**Comment by chrislloyd** [comment]
Fixed in #55
